test/test_nn.py

```diff
@@ -2958,9 +2958,11 @@ def compare_cpu_gpu(outputs_cpu, outputs_gpu):
                 grad_hy = torch.randn(num_layers * num_directions, batch, hidden_size)
 
                 if variable_len:
-                    batch_sizes = [7, 5, 5, 2, 1, 1]
-                    input_val = rnn_utils.pack_padded_sequence(input_val, batch_sizes, batch_first=batch_first)
-                    grad_output = rnn_utils.pack_padded_sequence(grad_output, batch_sizes, batch_first=batch_first).data
+                    lengths = torch.IntTensor([7, 5, 5, 2, 1, 1])
```

Comment:
The old code should continue to work even if we pass it a list directly, otherwise it's BC-breaking, right? Can we have a warning for when this use case happens?

Reply:
Concretely:

1. If some code exercises the externally facing API, as this code does, you should keep the test for the legacy API because someone might be using it, and we need to continue testing that it works.
2. You should add a warning to `pack_padded_sequence` that detects if you get a Python list and throws a deprecation notice.

Reply:
it is BC-breaking. I'll add the shim you've asked for below.